src/hooks/use-aggrid-local-row-filter.ts

```diff
@@ -90,14 +118,74 @@ export const useAggridLocalRowFilter = (
 
             // If all columns referenced by the filters exist, apply the filters
             if (allColumnsExist) {
+                const filterWithTolerance = addToleranceToFilter(filters);
                 // Format the filters for AG Grid and apply them using setFilterModel
-                const formattedFilters = formatCustomFiltersForAgGrid(filters);
+                const formattedFilters =
+                    formatCustomFiltersForAgGrid(filterWithTolerance);
                 gridRef.current.api.setFilterModel(formattedFilters);
             }
         },
         [formatCustomFiltersForAgGrid, gridRef]
     );
 
+    const addToleranceToFilter = (
+        filters: FilterSelectorType[],
+        tolerance: number = 0.00001
+    ): FilterSelectorType[] => {
+        return filters
+            .map((filter): FilterSelectorType | FilterSelectorType[] => {
+                // Attempt to convert filter value to a number if it's a string, otherwise keep it as is
+                let valueAsNumber: number | string[] =
+                    typeof filter.value === 'string'
+                        ? parseFloat(filter.value)
+                        : filter.value;
+                // If the value is successfully converted to a number, apply tolerance adjustments
+                if (typeof valueAsNumber === 'number') {
+                    let facteur = Math.pow(10, 5);
+                    // Truncate the number to maintain precision
+                    let nombre1Tronque =
+                        Math.floor(valueAsNumber * facteur) / facteur;
+                    // Depending on the filter type, adjust the filter value by adding or subtracting the tolerance
+                    switch (filter.type) {
+                        case 'notEqual':
+                            // For 'notEqual', create two conditions: one for greaterThan and one for lessThan
+                            return [
+                                {
+                                    ...filter,
+                                    type: 'greaterThan',
+                                    value: (nombre1Tronque + tolerance).toFixed(
+                                        5
```

Comment:
I think this 5 is also linked to tolerance, or at least to "facteur"
make them linked with a CONSTANT

Reply:
done